src/index.ts

```diff
@@ -183,39 +186,30 @@ class Logify<ErrorTypes = ErrorParamType> {
    * @private
    */
   private _sendLogToGrafana = async (log: string) => {
-    const hasLogicToSend = isBoolean(this.props.shouldSendLogsIf) || isFunction(this.props.shouldSendLogsIf)
+    const shouldSend = (): boolean => {
+      const hasLogicToSend = isBoolean(this.props.shouldSendLogsIf) || isFunction(this.props.shouldSendLogsIf)
+
+      if (!hasLogicToSend) {
+        return true
+      }
 
-    if (hasLogicToSend) {
-      const shouldSend = isBoolean(this.props.shouldSendLogsIf)
+      return isBoolean(this.props.shouldSendLogsIf)
         ? this.props.shouldSendLogsIf
         : // @ts-ignore
-          this.props?.shouldSendLogsIf?.()
-
-      if (shouldSend) {
-        try {
-          fetch(this.props.endpoint, {
-            method: 'POST',
-            body: log,
-            headers: {
-              'Content-Type': 'application/json; charset=utf-8',
-            },
-          })
-        } catch (e) {
-          this.debug('error sending log', e as Record<string, unknown>)
-        }
-      }
-    } else {
-      try {
-        fetch(this.props.endpoint, {
-          method: 'POST',
-          body: log,
-          headers: {
-            'Content-Type': 'application/json; charset=utf-8',
-          },
-        })
-      } catch (e) {
+        this.props?.shouldSendLogsIf?.()
+    }
+
+    if (shouldSend()) {
+      fetch(this.props.endpoint, {
+        method: 'POST',
+        body: log,
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8',
+        },
+      }).catch(e => {
         this.debug('error sending log', e as Record<string, unknown>)
-      }
+        this._store(log)
```

Comment:
NOTE:
With the current implementation, all logs that are failed to send (for whatever reason) will be stored. 
It would probably be better to only store the logs that failed to send because of:
a) Network connection missing
b) Access token missing

But missing access token is not a reproducable case as these requests are unauthorized right now.

Reply:
ok good to have in mind. i think we can move forward with this approach and come back to fix things based on what we'll observe.